packages/lesswrong/server/resolvers/jargonResolvers/jargonTermMutations.ts

```diff
@@ -0,0 +1,256 @@
+  // Regular expression to match aria-label attributes
+  const ariaLabelRegex = /aria-label="([^"]*)"/g;
+  
+  // Regular expression to match common LaTeX patterns
+  const latexPattern = /\\[a-zA-Z]+|[_^{}]|\$/;
+  
+  // Array to store results
+  const results: string[] = [];
+  
+  let match;
+  while ((match = ariaLabelRegex.exec(htmlContent)) !== null) {
+    const ariaLabel = match[1];
+    if (latexPattern.test(ariaLabel)) {
+      results.push(ariaLabel);
+    }
+  }
+  
+  return results.join(",");
+}
+
+
+const getMathExample = (() => {
+  let examplePostContents: string;
+
+  return async () => {
+    if (!examplePostContents) {
+      const pathName = './public/exampleMathPost.md'
+      examplePostContents = (await readFile(pathName)).toString()
+    }
+
+    return examplePostContents;
+  };
+})();
+
+
+export async function getLaTeXExplanations(post: PostsPage) {
+  const originalHtml = post.contents?.html ?? ""
+  const originalMarkdown = htmlToMarkdown(originalHtml)
+  const markdown = (originalMarkdown.length < 200 * 1000) ? originalMarkdown : originalMarkdown.slice(0, 200 * 1000)
+
+  const terms = identifyLatexTerms(originalHtml)
+  const exampleMathPost = await getMathExample();
+
+  const response = (await queryClaudeForJargonExplanations({markdown, terms, formatPrompt: mathFormatPrompt, examplePost: exampleMathPost, exampleExplanations: exampleMathGlossary})) || []
+
+  return response?.map(jargon => ({
+    term: jargon.term,
+    text: `<div>
+      <p>${jargon.text}</p>
+      <p><strong>Concrete Example:</strong> ${jargon.concreteExample}</p>
+      <p><strong>Why It Matters:</strong> ${jargon.whyItMatters}</p>
+      <p><em>Math Basics: ${jargon.mathBasics}</em></p>
+    </div>`,
+    altTerms: jargon.altTerms,
+  }))
+}
+
+export async function createEnglishExplanations(post: PostsPage) {
+  const originalHtml = post.contents?.html ?? ""
+  const originalMarkdown = htmlToMarkdown(originalHtml)
+  const markdown = (originalMarkdown.length < 200 * 1000) ? originalMarkdown : originalMarkdown.slice(0, 200 * 1000)
+
+  const terms = await queryClaudeForTerms(markdown)
+  if (!terms) return []
+
+  return queryClaudeForJargonExplanations({markdown, terms, formatPrompt, examplePost: exampleJargonPost2, exampleExplanations: exampleJargonGlossary2})
+}
+
+export const createNewJargonTerms = async (postId: string, currentUser: DbUser) => {
+  const post = await fetchFragmentSingle({
+    collectionName: 'Posts',
+    fragmentName: 'PostsPage',
+    currentUser,
+    selector: { _id: postId },
+  });
+  if (!post) {
+    throw new Error('Post not found');
+  }
+  // console.log("creatingLatexTerms")
+  // const newMathJargon = await getLaTeXExplanations(post) || []
+  const newEnglishJargon = await createEnglishExplanations(post) || []
+
+  console.log("creating jargonTerms")
+  const newJargonTerms = await Promise.all([
+    ...newEnglishJargon.map(term =>
+      createMutator({
```

Comment:
Using `createMutator` does just make it impossible to reliably avoid creating two sets of jargon terms if two calls to `createNewJargonTerms` get made at roughly the same time.  This should probably be a repo method doing a raw sql `INSERT INTO` with a `WHERE` clause that checks no jargon terms were created in the last [recent span of time].  That still isn't perfectly safe but should prevent the overwhelming majority of double-inserts.